modules/homa-lite/src/lib.rs

```diff
@@ -283,36 +373,505 @@ pub mod module {
 			T::GovernanceOrigin::ensure_origin(origin)?;
 
-			XcmDestWeight::<T>::put(xcm_dest_weight);
-			Self::deposit_event(Event::<T>::XcmDestWeightSet(xcm_dest_weight));
+			XcmBaseWeight::<T>::put(xcm_base_weight);
+			Self::deposit_event(Event::<T>::XcmBaseWeightSet(xcm_base_weight));
+			Ok(())
+		}
+
+		/// Sets the xcm_unbond_fee for XCM transaction. Fee is paid in Staking Currency.
+		/// Requires `T::GovernanceOrigin`
+		///
+		/// Parameters:
+		/// - `xcm_unbond_fee`: The new fee for XCM transfers.
+		#[pallet::weight(< T as Config >::WeightInfo::set_xcm_unbond_fee())]
+		#[transactional]
+		pub fn set_xcm_unbond_fee(origin: OriginFor<T>, xcm_unbond_fee: Balance) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			XcmUnbondFee::<T>::put(xcm_unbond_fee);
+			Self::deposit_event(Event::<T>::XcmUnbondFeeSet(xcm_unbond_fee));
+			Ok(())
+		}
+
+		/// Put in an request to redeem Staking currencies used to mint Liquid currency.
+		/// The redemption will happen after the currencies are unbonded on the relaychain.
+		///
+		/// Parameters:
+		/// - `liquid_amount`: The amount of liquid currency to be redeemed into Staking currency.
+		/// - `additional_fee`: Percentage of the fee to be awarded to the minter.
+		#[pallet::weight(< T as Config >::WeightInfo::request_redeem())]
+		#[transactional]
+		pub fn request_redeem(origin: OriginFor<T>, liquid_amount: Balance, additional_fee: Permill) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			if liquid_amount.is_zero() {
+				// If the maount is zero, cancel previuos redeem request.
+				if let Some((request_amount, _)) = Self::redeem_requests(&who) {
+					// Unreserve the liquid fee and remove the redeem request.
+					let unreserved = T::Currency::unreserve(T::LiquidCurrencyId::get(), &who, request_amount);
+					ensure!(unreserved.is_zero(), Error::<T>::InsufficientReservedBalances);
+					RedeemRequests::<T>::remove(&who);
+
+					Self::deposit_event(Event::<T>::RedeemRequestCancelled(who, request_amount));
+				}
+			} else {
+				// If there are available_staking_balances, redeem immediately with no additional fee.
+				let mut available_staking_balance = Self::available_staking_balance();
+				let actual_liquid_amount = min(
+					liquid_amount,
+					Self::convert_staking_to_liquid(available_staking_balance)?,
+				);
+
+				if !actual_liquid_amount.is_zero() {
+					// Immediately redeem from the available_staking_balances
+					let actual_staking_amount = Self::convert_liquid_to_staking(actual_liquid_amount)?;
+
+					// Redeem from the available_staking_balances costs no extra fee.
+					T::Currency::deposit(T::StakingCurrencyId::get(), &who, actual_staking_amount)?;
+					let slash_amount = T::Currency::slash(T::LiquidCurrencyId::get(), &who, actual_liquid_amount);
+					ensure!(slash_amount.is_zero(), Error::<T>::InsufficientLiquidBalance);
+
+					// Update the available_staking_balance
+					available_staking_balance = available_staking_balance
```

Comment:
```suggestion
					let available_staking_balance = available_staking_balance
```
and no need to make it `let mut` on line 429